k8s_test_harness/util/platform_util.py

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright 2024 Canonical, Ltd.
+# See LICENSE file for licensing details
+#
+
+"""Module containing utilities for querying platform parameters
+for ROCK testing."""
+
+import platform
+
+
+ROCKCRAFT_PLATFORM_AMD64 = "amd64"
+ROCKCRAFT_PLATFORM_I386 = "i386"
+ROCKCRAFT_PLATFORM_ARM64 = "arm64"
+ROCKCRAFT_PLATFORM_ARMHF = "armhf"
+ROCKCRAFT_PLATFORM_S390X = "s390x"
+ROCKCRAFT_PLATFORM_PPC64EL = "ppc64el"
+ROCKCRAFT_PLATFORM_RISCV64 = "riscv64"
+
+# NOTE(aznashwan): these constants are automatically derived at runtime with
+# no static definition for them, so we must maintain the definitions ourselves:
+PYTHON_MACHINE_X86 = "x86"
+PYTHON_MACHINE_X86_64 = "x86_64"
+PYTHON_MACHINE_ARM64 = "arm64"
+
+
+_PYTHON_MACHINE_TO_ROCKCRAFT_PLATFORM_ARCHITECTURE_MAP = {
```

Comment:
nit: let's just add what we need. 

```
PYTHON_TO_ROCKCRAFT_ARCH_MAP = {
    "x86": "i386",
    ...

}
```
you don't use `ROCKCRAFT_PLATFORM_PPC64EL` etc. so let's don't define them.

Reply:
Adding onto this, can we also make this a bit more generic instead of `ROCKCRAFT`, this harness can be re-used in other projects and a platform/arch check seems applicable to things other than rocks such as snaps etc.

Reply:
@berkayoz you're right, but I don't know whether it is guaranteed that the label set will be the same for non-`rockcraft` projects. ([`rockcraft` apparently follows Golang's platform labelling scheme](https://github.com/canonical/rockcraft/blob/08e9a1dec8fd20ebea56a841bcfab1630bb13e1c/rockcraft/architectures.py#L25-L62), for example)

At the end of the day we will have to pick which set of arch labels we want to go for (pun not intended), which is why I wanted to make both the constants/function's name explicitly indicate it.

I can change the naming if you insist, just let me know.

Reply:
agreed